ValveKeyValue/ValveKeyValue/Deserialization/KV1TokenReader.cs

```diff
@@ -84,12 +85,34 @@ KVToken ReadComment()
         {
             ReadChar(CommentBegin);
 
-            if (Peek() == (char)CommentBegin)
+            var sb = new StringBuilder();
+            var next = Next();
+            
+            // Some keyvalues implementations have a bug where only a single slash is needed for a comment
+            if (next != CommentBegin)
             {
-                Next();
+                sb.Append(next);
+            }
+
+            while (true)
+            {
+                next = Next();
+
+                if (next == '\n')
+                {
+                    break;
+                }
+
+                sb.Append(next);
+            }
+
+            if (sb[^1] == '\r')
+            {
+                sb.Remove(sb.Length - 1, 1);
             }
 
-            var text = textReader.ReadLine();
+            var text = sb.ToString();
```

Comment:
Why this loop? Is there something wrong with `textReader.ReadLine()` from the old code? If its the case that we might have something in our little `peekedNext` field, can we not deal with that in a simpler manner?

Reply:
I wonder if it would be simpler to build a `TextReader` wrapper.